pkgs/leak_tracker_flutter_testing/test/tests/leak_tracking_for_tests_test.dart

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:leak_tracker/leak_tracker.dart';
+import 'package:leak_tracker_flutter_testing/src/leak_testing.dart';
+
+class _Classes {
+  static const anyLeak1 = 'anyLeak1';
+  static const notDisposed1 = 'notDisposed1';
+  static const notGCed1 = 'notGCed1';
+
+  static const anyLeak2 = 'anyLeak2';
+  static const notDisposed2 = 'notDisposed2';
+  static const notGCed2 = 'notGCed2';
+
+  static const anyLeak3 = 'anyLeak3';
+  static const notDisposed3 = 'notDisposed3';
+  static const notGCed3 = 'notGCed3';
+
+  static final all = [
+    anyLeak1,
+    anyLeak2,
+    anyLeak3,
+    notDisposed1,
+    notDisposed2,
+    notDisposed3,
+    notGCed1,
+    notGCed2,
+    notGCed3,
+  ];
+
+  static List<String> others(List<String> classes) =>
+      all.where((c) => !classes.contains(c)).toList();
+}
+
+/// Returns true, if the provided [classes] are skipped and all other classes from [_Classes] are tracked.
+bool _areOnlySkipped(
+  List<String> classes, {
+  LeakType? leakType,
+  LeakTesting? settings,
+}) {
+  final theSettings = settings ?? LeakTesting.settings;
+  final classesSkipped = !classes
+      .map(
+        (theClass) =>
+            theSettings.ignoredLeaks.isIgnored(theClass, leakType: leakType),
+      )
+      .any((skipped) => !skipped);
+  final othersTracked = _Classes.others(classes)
+      .map(
+        (theClass) =>
+            theSettings.ignoredLeaks.isIgnored(theClass, leakType: leakType),
+      )
+      .any((skipped) => !skipped);
+  return classesSkipped && othersTracked;
+}
+
+void main() {
+  setUp(() {
+    LeakTesting.settings = const LeakTesting();
+  });
```

Comment:
Could you do the following and then remove `@visibleForTesting` from the LeakTesting constructor and make it actually private to underline that nobody should be instantiating it directly?

```suggestion
  late LeakTesting defaults;

  setUp(() {
    defaults = LeakTesting.settings;
  });
  tearDown(() {
    LeakTesting.settings = defaults;
  });
```